### ISSUE TITLE
How to reproduce the paper metrics?

### ISSUE THREAD

**Issue by CristhianBoujon** [megagonlabs/ditto#4]
I run the suggested example in the readme to try reproduce results showed in the [paper](https://arxiv.org/pdf/2004.00584v2.pdf):
```
CUDA_VISIBLE_DEVICES=0 python train_ditto.py \
  --task Structured/Beer \
  --batch_size 64 \
  --max_len 64 \
  --lr 3e-5 \
  --n_epochs 40 \
  --finetuning \
  --lm distilbert \
  --fp16 \
  --da del \
  --dk product \
  --summarize
```

If I'm right, based on the paper I expected to get around F1-score around 94.7
![Captura de pantalla 2020-09-03 a la(s) 16 24 33](https://user-images.githubusercontent.com/1195235/92158616-92e44480-ee02-11ea-9742-0f3fb6280a4d.png)

But I just get: 
<img width="328" alt="Captura de pantalla 2020-09-03 a la(s) 16 36 53" src="https://user-images.githubusercontent.com/1195235/92159420-be1b6380-ee03-11ea-847e-3017a69f2349.png">

You can also see the training logs  [here](https://github.com/megagonlabs/ditto/files/5170805/training_output.txt)

What am I doing wrong?

**Comment by oi02lyl** [comment]
For this dataset, we use the following setting (the summarization option is off). We used roberta, the drop_col da operator, the general dk, and a batch size of 32 for this dataset. The command should look like this:

`CUDA_VISIBLE_DEVICES=0 python train_ditto.py \
  --task Structured/Beer \
  --batch_size 32 \
  --max_len 256 \
  --lr 3e-5 \
  --n_epochs 40 \
  --finetuning \
  --lm roberta \
  --fp16 \
  --da drop_col \
  --dk general`

I just re-ran the experiment and here are the results:

**Baseline (no dk or da)**

=========eval at epoch=40=========
Validation:
=============Structured/Beer==================
accuracy=0.967
precision=0.867
recall=0.929
f1=0.897
======================================
Test:
=============Structured/Beer==================
accuracy=0.967
precision=0.824
recall=1.000
f1=0.903
======================================

**With DK (general) only** 
Somehow the test result is the same as the baseline.

=========eval at epoch=40=========
Validation:
=============Structured/Beer==================
accuracy=0.967
precision=0.824
recall=1.000
f1=0.903
======================================
Test:
=============Structured/Beer==================
accuracy=0.967
precision=0.824
recall=1.000
f1=0.903
======================================


**Ditto (with both dk and da)** 

=========eval at epoch=40=========
Validation:
=============Structured/Beer==================
accuracy=0.956
precision=0.857
recall=0.857
f1=0.857
======================================
Test:
=============Structured/Beer==================
accuracy=0.978
precision=0.875
recall=1.000
f1=0.933
======================================

In the paper's experiments, we reported the results by taking the epoch with the highest validation F1 score (here I am taking the last epoch).

I did notice something strange as the baseline result seems significantly higher than before. I remember that we updated the MixDA code from snippext which might have some effect on the data augmentation results too.

**Comment by CristhianBoujon** [comment]
Thank you!
